public/app/plugins/panel/canvas/InlineEdit.tsx

```diff
@@ -0,0 +1,121 @@
+import { css, cx } from '@emotion/css';
+import React, { SyntheticEvent, useRef, useState } from 'react';
+import Draggable from 'react-draggable';
+import { Resizable, ResizeCallbackData } from 'react-resizable';
+
+import { Dimensions2D, GrafanaTheme2 } from '@grafana/data';
+import { IconButton, Portal, useStyles2 } from '@grafana/ui';
+import store from 'app/core/store';
+
+import { InlineEditOptions } from './InlineEditOptions';
+
+type Props = {
+  onClose?: () => void;
+};
+
+const OFFSET_X = 70;
+
+export const InlineEdit = ({ onClose }: Props) => {
+  const btnInlineEdit = document.querySelector('[data-btninlineedit]')!.getBoundingClientRect();
+  const ref = useRef<HTMLDivElement>(null);
+  const styles = useStyles2(getStyles);
+  const inlineEditKey = 'inlineEditPanel';
+
+  const defaultMeasurements = { width: 350, height: 400 };
+  const defaultX = btnInlineEdit.x + OFFSET_X;
+  const defaultY = btnInlineEdit.y - defaultMeasurements.height;
+
+  const savedPlacement = store.getObject(inlineEditKey, {
+    x: defaultX,
+    y: defaultY,
+    w: defaultMeasurements.width,
+    h: defaultMeasurements.height,
+  });
+  const [measurements, setMeasurements] = useState<Dimensions2D>({ width: savedPlacement.w, height: savedPlacement.h });
+  const [placement, setPlacement] = useState({ x: savedPlacement.x, y: savedPlacement.y });
+
+  const onDragStop = (event: any, dragElement: any) => {
+    let x = dragElement.x < 0 ? 0 : dragElement.x;
+    let y = dragElement.y < 0 ? 0 : dragElement.y;
+
+    setPlacement({ x: x, y: y });
+    saveToStore(x, y, measurements.width, measurements.height);
+  };
+
+  const onResizeStop = (event: SyntheticEvent<Element, Event>, data: ResizeCallbackData) => {
+    const { size } = data;
+    setMeasurements({ width: size.width, height: size.height });
+    saveToStore(placement.x, placement.y, size.width, size.height);
+  };
+
+  const saveToStore = (x: number, y: number, width: number, height: number) => {
+    store.setObject(inlineEditKey, { x: x, y: y, w: width, h: height });
+  };
+
+  return (
+    <Portal>
+      <div className={styles.draggableWrapper}>
+        <Draggable handle="strong" onStop={onDragStop} position={{ x: placement.x, y: placement.y }}>
+          <Resizable height={measurements.height} width={measurements.width} onResize={onResizeStop}>
+            <div
+              className={cx('box', 'no-cursor', `${styles.inlineEditorContainer}`)}
```

Comment:
wrapping ${styles.inlineEditorContainer} in a string literal template is not doing anything here :)